flow/project.py

```diff
@@ -803,18 +897,30 @@ def isdisjoint(self, group):
         """
         return set(self).isdisjoint(set(group))
 
-    def _generate_id(self, job, operation_name=None, index=0):
+    def calc_id_str(self, jobs, upto=None):
+        jobstr = ''
+        length_not_defined=False
+        if upto is None:
+            length_not_defined=True
+        for job in jobs:
+            if length_not_defined:
+                upto = len(str(job.id))
+            jobstr+=str(job.id)[:upto]+'-'
+        return jobstr.strip('-')
+
+    def _generate_id(self, jobs, operation_name=None, index=0):
         "Return an id, which identifies this group with respect to this job."
-        project = job._project
+        project = jobs[0]._project
```

Comment:
We're assuming that the jobs are from the same project. When we're done executing this, we filter the JobOperation instance and there we check whether every job is in the same project or not. Though it seems to me that I should test whether the jobs are in the same project or not here also.

Reply:
Might be good to make that assumption explicit with an assert. This could be a significant performance drag, but now is not the time for optimization, now is the time for code safety.

Reply:
That's true, will insert assertion

Reply:
Eventually, I think this can be covered by tests, but for now safety is the right approach.

Reply:
does this mean that if two aggregates have the same job first in their lists that their ids would be identical?

Reply:
@atravitz 
This can be a problem if we run in something like this `python project.py run -o op_group_of_2 -j J1 J1 J1 J1`
This behaviour seems fair to me because even if we generate similar ids, we'll just be overwriting the `id` in the project document.